src/resources/schema/document-code.yml

```diff
@@ -31,8 +31,10 @@
 
       Code linking is currently implemented only for the knitr engine 
       (via the [downlit](https://downlit.r-lib.org/) package). 
-      A limitation of downlit currently prevents code linking 
-      if `code-line-numbers` is also `true`.
+      A limitation of downlit prevents code linking 
+      if `code-line-numbers` and/or `code-annotations` are set to `true`.
+      For `code-link` to work properly, `code-line-numbers` and `code-annotations`
+      require to be set to `false`.
```

Comment:
Don't know if you tried but slight side effect.... setting `code-annotations: false` does also prevent code linking it seems if the annotations are present in the chunk. 

Basically, if one wants code linking in a cells, it should not contains any annotation using `# <1>` syntax. 

I don't know why `code-annotations: false` make code linking not work because I would have thought the HTML would not be changed in that case. (which is the underlying issue with downlit)

I would say that the advice would be 

* For `code-link` to work properly, `code-line-numbers` should not be activated on the cell and   No code annotations syntax added.  

But again, probably a bug that `code-annotations` still do something... unless this is the `# <1>` syntax that makes **downlit** fails... 🤔 

Are you observing like me ?

Reply:
FYI, I made a repository to check various combinations (but only when defining the option in the yaml frontmatter): https://github.com/mcanouil/quarto--code-link--code-annotations

Reply:
After checking, code linking is not preventing by the use of code annotations, it is the other way around as I initially described, _i.e._, code annotation does not work properly when `code-linking: true`.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/a496afb8-53b7-467a-922d-f5eca5f0b232">

Reply:
FYI, for `code-line-numbers`, see <https://github.com/mcanouil/quarto--code-link--code-line-numbers>.
This indeed prevents code linking.

Reply:
Really weird - I can indeed reproduce and see that code links works in your example, but I maintain that in my previous example it still does not. Did you try ?  https://github.com/quarto-dev/quarto-cli/pull/6297#discussion_r1270405944

Simplified example 
````markdown
---
format: 
  html:
    code-link: true
    code-annotations: true
---

```r
stats::rnorm(1) # <1>
stats::rnorm(2) # <2>
```

1. Test 1
2. Test 2 

````

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/3aaf7614-c5d5-4477-819a-aa0622c0f884)


So we should definitely try every possible case and check how that works - for now I don't understand why this would work in the above and not in your case. Looking into it. 

Thanks for the repo by the way !

Reply:
Sorry it has been a while. Do we need to check more cases for theses options before merging ?

Reply:
I have to say, indeed it's been a while and I did not update https://github.com/mcanouil/quarto--code-link--code-line-numbers with your last example.
I'll have to add those tests and check the results.